compiler/modules/front/const.bal

```diff
@@ -517,47 +520,86 @@ function foldVarRefExpr(FoldContext cx, t:SemType? expectedType, s:VarRefExpr ex
     }
 }
 
-function foldFloatLiteralExpr(FoldContext cx, t:SemType? expectedType, s:FpLiteralExpr expr) returns s:ConstValueExpr|FoldError {
-    // This will need to change when we support decimal
-    float|error result = floatFromDecimalLiteral(expr.untypedLiteral);
-    if result is float {
-        return { startPos: expr.startPos, endPos: expr.endPos, value: result };
+function foldFpLiteralExpr(FoldContext cx, t:SemType? expectedType, s:FpLiteralExpr expr) returns s:ConstValueExpr|FoldError { 
+    float|decimal result;
+    t:Context tc = cx.typeContext();
+    s:FpTypeSuffix? typeSuffix = expr.typeSuffix;
+    string untypedLiteral = expr.untypedLiteral;
+    Position startPos = expr.startPos;
+    if typeSuffix != () {
```

Comment:
- let's use destructure
- move `result` last since it's a no-init decl
```suggestion
    var { typeSuffix, untypedLiteral, startPos } = expr.typeSuffix;
    t:Context tc = cx.typeContext();
    float|decimal result;
    if typeSuffix != () {
```

Reply:
Changed